### ISSUE TITLE
gpio.test test failures on more recent platforms

### ISSUE THREAD

**Issue by gallex** [warthog618/gpio#4]
run  gpop.test  sometimes  OK,  sometimes may FAIL. 

error message is following: 

pi@raspberrypi:~ $ ./gpio.test -test.bench=.*
--- FAIL: TestCloseInterrupts (0.06s)
    interrupt_test.go:312: Spurious interrupt during close 0
FAIL

**Comment by warthog618** [comment]
Have you tied J8p15 and J8p16 as per the documentation?

**Comment by gallex** [comment]
Yes, I  tied J8p15 and J8p16 with a jumper

**Comment by warthog618** [comment]
Any other clues?  Platform?

**Comment by gallex** [comment]
3B+  sometimes OK, sometimes may FAIL

4B always FAIL

**Comment by warthog618** [comment]
Haven't tested on a 3B - don't have one.
Just received a 4B today as it happens, and will try it out presently.

**Comment by warthog618** [comment]
4B mostly works for me.  I get a PullDown failure, which is probably due to the faster CPU - so the line isn't given time to reach its new equilibrium level before the test checks it.  You are probably getting something similar with ringing on the the interrupt line.

I'll have a look and see if I can make the tests more robust on the faster hardware.

In the meantime, if you aren't toggling lines at full speed like the tests do, you should be fine.

**Comment by gallex** [comment]
I'll try , thank you.

**Comment by gallex** [comment]
I get a PullDown failure, which is probably due to the faster CPU
----------

I add a time.Sleep(n*time.Millisecond) , even n =100,  get a PullDown failure also.

**Comment by warthog618** [comment]
How pullups are programmed has changed in the Pi4, so that is currently broken.  The line that the tests use happens to be pulled up by default, so pull up appears to work, while pull down fails.  Will have to look into what needs to be changed, but this is not a high priority for me at the moment.

Other than the pull downs the tests are totally solid for me on all the the platforms I have available so not sure what I can change, or in fact if the problems don't lie in your boards.

**Comment by warthog618** [comment]
I've committed a change to handle the Raspberry Pi 4.
I've also fixed some race conditions in the watcher.Close and TestCloseInterrupts that were showing up on the faster platforms.
All tests are now consistently passing for me on all platforms.
I've also added a new command line utility, which unfortunately for you will pull in even more dependencies.
I've got a few more minor changes to add, and then I'll tag a new release in the next few days.
